Plugins/Public/playercntl_plugin/MiscCmds.cpp

```diff
@@ -252,6 +265,184 @@ namespace MiscCmds
+		{
+			for (map<wstring, uint>::iterator i = tempmap.begin();
+				i != tempmap.end(); ++i)
+			{
+				////This is intended  to write to disk only time limits which have not expired yet
+				if (((int)time(0) - i->second) < set_iResetrepTimeLimit)
+				{
+					wstring temp = i->first;
+					fprintf_s(file, "[Playership]\n");
+					ini_write_wstring(file, "shipname", temp);
+					fprintf_s(file, "lastresettime = %d\n", i->second);
+				}
+			}
+			fclose(file);
+		}
+	}
+
+	/** A command to reset all reputations to zero that you might have */
+	bool MiscCmds::UserCmd_ResetRep(uint iClientID, const wstring &wscCmd, const wstring &wscParam, const wchar_t *usage)
+	{
+		HK_ERROR err;
+
+		wstring wscCharname = (const wchar_t*)Players.GetActiveCharacterName(iClientID);
+
+		uint iShip;
+		pub::Player::GetShip(iClientID, iShip);
+		if (iShip)
+		{
+			PrintUserCmdText(iClientID, L"ERR Not in base");
+			return true;
+		}
+
+		// Read the current number of credits for the player
+		// and check that the character has enough cash.
+		int iCash = 0;
+		if ((err = HkGetCash(wscCharname, iCash)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR Resetrep failed to read player cash");
+			return true;
+		}
+
+		if (set_iResetrepCost > 0 && iCash < set_iResetrepCost)
+		{
+			PrintUserCmdText(iClientID, L"ERR Insufficient credits");
+			return true;
+		}
+
+
+		wstring wscCharFileName;
+		if ((err = HkGetCharFileName(wscCharname, wscCharFileName)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR Resetrep failed to get charfilename");
+			return true;
+		}
+
+		// Read the last time a rename was done on this character
+		wstring wscDir;
+		if ((err = HkGetAccountDirName(wscCharname, wscDir)) != HKE_OK)
+		{
+			PrintUserCmdText(iClientID, L"ERR Resetrep failed to get account dir");
+			return true;
+		}
+		string scRenameFile = scAcctPath + wstos(wscDir) + "\\" + "resetrep.ini";
+		map<wstring, uint> CharfilesLastResetRepUsage = Resetrep_load_Time_limits_for_player_account(scRenameFile);
+		if (CharfilesLastResetRepUsage.find(wscCharFileName) != CharfilesLastResetRepUsage.end())
+		{
+			uint LastTime = CharfilesLastResetRepUsage[wscCharFileName];
+
+			if (((int)time(0) - LastTime) < set_iResetrepTimeLimit)
+			{
+				int SecondsLeft = (set_iResetrepTimeLimit - ((int)time(0) - int(LastTime)));
+				int DaysLeft = SecondsLeft / (3600 * 24);
+				PrintUserCmdText(iClientID, L"ERR Attempt to resetrep too frequently. Please wait %d days (%d in seconds).", DaysLeft, SecondsLeft);
+				return true;
+			}
+		}
+
+		int count = 0;
+		for (map<string, uint>::iterator iter = factions.begin(); iter != factions.end(); iter++)
+		{
+			float fRep = 0.0f;
+			string factionName = iter->first;
+			HK_ERROR error;
+			if ((error = HkGetRep(wscCharname, stows(factionName), fRep)) != HKE_OK)
+			{
+				PrintUserCmdText(iClientID, L"ERR Resetrep, check for faction failed");
+				continue;
+			}
+			count++;
+			HkSetRep(wscCharname, stows(factionName), 0.0f);
```

Comment:
just sets all factions to neutral? shall we leave a comment saying setting all factions to neutral is safe because the player must be docked and rep hacks will take effect on undock?

Reply:
All right.

Reply:
Added
//We reset reps to zero, but it is safe because the ship is docked and kicked
//So rephacks take effect right after logging back